### PR TITLE
trigger MacOS builds from AWS

### PR DESCRIPTION
**Pull request by jjergus**
This built yesterday's release and nightly.

Code that runs on Azure is in this homebrew-hhvm branch: https://github.com/hhvm/homebrew-hhvm/tree/build-triggers

I don't know why everything conveniently fails exactly when I need to test error handling but check out how the Azure build [failed and got automatically retried](https://dev.azure.com/hhvm-oss/hhvm-oss-builds/_build?definitionId=1).

### CODE REVIEW

File: aws/hhvm1/lambdas/check_if_repos_changed.py [hunk 1]
```diff
@@ -16,6 +16,7 @@ def lambda_handler(event, context=None):
     if (
       'success' in vr.get('PublishBinaryPackages', {}) or
       'success' in vr.get('PublishSourceTarball', {}) or
+      'success' in vr.get('BuildAndPublishMacOS', {}) or
```
Comment (jjergus) [inline aws/hhvm1/lambdas/check_if_repos_changed.py]: this means we now also run the update indices + invalidate cloudfront steps after Mac stuff is published, I think that's correct...?

File: aws/hhvm1/lambdas/common.py [hunk 2]
```diff
@@ -11,13 +11,20 @@
 from urllib import request
 
 class Config:
-  override_org = None
-  override_branch = None
+  override_org = 'jjergus'
+  override_branch = 'macos'
```
Comment (jjergus) [inline aws/hhvm1/lambdas/common.py]: note to self: remove and deploy before merging

File: aws/hhvm1/state-machine/generate.hack [hunk 3]
```diff
@@ -327,19 +340,19 @@ function linear_state_machine(
  * Adds in the Succes/Failure states that we need in every nested state machine.
  */
 function nested_state_machine(
-  string $prefix,
-  ?P $add_param,
+  string $end_state_prefix,
+  ?P $add_param_to_results,
```
Comment (jjergus) [inline aws/hhvm1/state-machine/generate.hack]: I forgot what these were for and I only wrote it, like, last week, so I figured I should give them better names.

File: aws/hhvm1/state-machine/generate.hack [hunk 4]
```diff
@@ -348,6 +361,89 @@ function nested_state_machine(
   );
 }
 
+/**
+ * The two main parallel build branches.
+ */
+function linux_branch(): StateMachine {
+  return nested_state_machine(
```
Comment (jjergus) [inline aws/hhvm1/state-machine/generate.hack]: moved here without changes

File: aws/hhvm1/lambdas/check_if_repos_changed.py (hunk 1 above)
Comment (fredemmott) [reply aws/hhvm1/lambdas/check_if_repos_changed.py]: yep, that's a good thing.

Should probably have this triggered by a cloudwatch rule watching the S3 bucket instead...

File: aws/userdata/trigger-macos-builds.sh [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  | ssh-add -
+
+# Push a new commit to trigger the MacOS build.
+git clone git@github.com:hhvm/homebrew-hhvm.git
+cd homebrew-hhvm
+git config user.name "HHVM Homebrew Bot (AWS)"
+git config user.email opensource+hhvm-homebrew-bot@fb.com
+git checkout build-triggers
+
+TRIGGER_FILE="builds/$(date +%Y-%m-%d_%H-%M-%S)_$RANDOM.sh"
```
Comment (fredemmott) [inline aws/userdata/trigger-macos-builds.sh]: - Do these get cleaned up?
- looks like the content matters; if not, `git commit --allow-empty` would be nicer

Comment (jjergus) [reply aws/userdata/trigger-macos-builds.sh]: > content matters

Yeah this is how we pass information (which version to build etc.) from AWS to Azure.

> Do these get cleaned up?

Not yet...there's probably some value in keeping a history of what builds were scheduled with what parameters, but we should maybe try to clean up older ones.

**Pull request merged**